lib/logstash/filters/jdbc_static.rb

```diff
@@ -0,0 +1,205 @@
+  # These will usually be the definitions to setup the local in-memory tables.
+  # For example:
+  # local_db_objects => [
+  #   {type => "table", name => "servers", columns => [["id", "INTEGER"], ["ip", "varchar(64)"], ["name", "varchar(64)"], ["location", "varchar(64)"]]},
+  #   {type => "index", name => "servers_idx", table => "servers", columns => ["ip"]}
+  # ]
+  # NOTE: Important! Tables created here must have the same names as those used in the `loaders` and
+  # `local_lookups` configuration options
+  config :local_db_objects, :required => false, :default => [], :validate => [LogStash::Filters::Jdbc::DbObject]
+
+  # Define the list (Array) of enhancement local_lookups to be applied to an event
+  # Each entry is a hash of the query string, the target field and value and a
+  # parameter hash. Target is overwritten if existing. Target is optional,
+  # if omitted the lookup results will be written to the root of the event like this:
+  # event.set(<column name (or alias)>, <column value>)
+  # Use parameters to have this plugin put values from the event into the query.
+  # The parameter maps the symbol used in the query string to the field name in the event.
+  # NOTE: when using a query string that includes the LIKE keyword make sure that
+  # you provide a Logstash Event sprintf pattern with added wildcards.
+  # For example:
+  # local_lookups => [
+  #   {
+  #     "query" => "select * from country WHERE code = :code",
+  #     "parameters" => {"code" => "country_code"}
+  #     "target" => "country_details"
+  #   },
+  #   {
+  #     "query" => "select ip, name from servers WHERE ip LIKE :ip",
+  #     "parameters" => {"ip" => "%{[response][ip]}%"}
+  #     "target" => "servers"
+  #   }
+  # ]
+  config :local_lookups, :required => true, :validate => [LogStash::Filters::Jdbc::Lookup]
+
+  # Schedule of when to periodically run loaders, in Cron format
+  # for example: "* * * * *" (execute query every minute, on the minute)
+  #
+  # There is no schedule by default. If no schedule is given, then the loaders are run
+  # exactly once.
+  config :loader_schedule, :validate => [LogStash::Filters::Jdbc::LoaderSchedule]
+
+  # Append values to the `tags` field if sql error occured
+  config :tag_on_failure, :validate => :array, :default => ["_jdbcstreamingfailure"]
+
+  # Append values to the `tags` field if no record was found and default values were used
+  config :tag_on_default_use, :validate => :array, :default => ["_jdbcstreamingdefaultsused"]
+
+  # Remote Load DB Jdbc driver library path to third party driver library.
+  config :jdbc_driver_library, :validate => :path
+
+  # Remote Load DB Jdbc driver class to load, for example "oracle.jdbc.OracleDriver" or "org.apache.derby.jdbc.ClientDriver"
+  config :jdbc_driver_class, :validate => :string, :required => true
+
+  # Remote Load DB Jdbc connection string
+  config :jdbc_connection_string, :validate => :string, :required => true
```

Comment:
This string could contain credentials, it possible to define password in it, We should use this type https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/util/safe_uri.rb to protect it.

Reply:
The danger is to bleed that information in the log.

Reply:
SafeURI will not work as is. I would need to wrap compose it in a SafeConnectionString class because the jdbc bit in `jdbc:<possible uri>` needs to be stripped and even then the user password bits vary so much across DB technology.

MariaDB uses `jdbc:mariadb://localhost:3306/DB?user=root&password=myPassword`
MySql uses many:
`jdbc:mysql:loadbalance://localhost:3306,localhost:3310/sakila`
`jdbc:mysql://address=(protocol=tcp)(host=primaryhost)(port=3306),address=(protocol=tcp)(host=secondaryhost1)(port=3310)(user=test2)/test`
`jdbc:mysql://sandy:secret@[myhost1:1111,myhost2:2222]/db`
Others use properties that appear after a `;` not a `?`
MSSql uses `jdbc:sqlserver://localhost:1433;databaseName=AdventureWorks;user=UserName;password=*****`
Postgres uses `jdbc:postgresql://localhost/test?user=fred&password=secret&ssl=true`

I don't think SafeURI is gonna protect much.

Reply:
made perfect sense here, I guess its up to the secret store to protect theses.